app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,202 @@
+const outerHeight = (el) => {
+  let height = el.offsetHeight
+  const style = window.getComputedStyle(el)
+
+  height += parseInt(style.marginTop) + parseInt(style.marginBottom)
+  return height
+}
+
+class CheckboxSearchFilter {
+  constructor (params) {
+    this.params = params
+    this.container = params.container
+    this.container.classList.add('app-checkbox-filter--enhanced')
+    this.checkboxes = this.container.querySelectorAll("input[type='checkbox']")
+    this.checkboxesContainer = this.container.querySelector('.app-checkbox-filter__container')
+    this.checkboxesInnerContainer = this.checkboxesContainer.querySelector('.app-checkbox-filter__container-inner')
+    this.setup()
+  }
+
+  filterCheckboxes () {
+    const textValue = this.sanitizeText(this.textBox.value)
+    const allCheckboxes = this.getAllCheckboxes()
+
+    for (let i = 0; i < allCheckboxes.length; i++) {
+      const labelValue = this.sanitizeText(allCheckboxes[i].querySelector('.govuk-checkboxes__label').textContent)
+      if (labelValue.search(textValue) === -1) {
+        allCheckboxes[i].style.display = 'none'
+      } else {
+        allCheckboxes[i].style.display = 'block'
+      }
+    }
+
+    this.updateStatusBox({
+      foundCount: this.getAllVisibleCheckboxes().length,
+      checkedCount: this.getAllVisibleCheckedCheckboxes().length
+    })
+  }
+
+  getAllCheckboxes () {
+    return this.checkboxesContainer.querySelectorAll('.govuk-checkboxes__item')
+  }
+
+  getAllVisibleCheckboxes () {
+    return [].filter.call(this.getAllCheckboxes(), function (el) {
```

Comment:
Have you checked IE8 for example that Array.prototype.filter is being polyfilled?